spec/unit/provider/package/cab_spec.rb

```diff
@@ -126,6 +126,61 @@ def allow_get_packages
     end
   end
 
+  describe "#source_resource" do
+    before do
+      new_resource.source = "https://www.something.com/Test6.1-KB2664825-v3-x64.cab"
+      new_resource.cookbook_name = "cab_package"
+    end
+
+    it "sets the desired parameters of downloades cab file" do
+      allow(provider).to receive(:default_download_cache_path).and_return("C:\\chef\\cache\\package")
+      source_resource = provider.source_resource
+      expect(source_resource.path).to be == "C:\\chef\\cache\\package"
+      expect(source_resource.name).to be == "windows_test_pkg"
+      expect(source_resource.source).to be == [new_resource.source]
+      expect(source_resource.cookbook_name).to be == "cab_package"
+    end
+  end
+
+  describe "#default_download_cache_path" do
+    before do
+      new_resource.source = "https://www.something.com/Test6.1-KB2664825-v3-x64.cab"
+    end
+
+    it "returns a clean cache path where the cab file is downloaded" do
+      allow(Chef::FileCache).to receive(:create_cache_path).and_return(ENV["TEMP"])
+      path = provider.default_download_cache_path
+      if windows?
+        expect(path).to be == File.join("#{ENV['TEMP']}", "\\", "Test6.1-KB2664825-v3-x64.cab")
+      else
+        expect(path).to be == File.join("#{ENV['TEMP']}", "Test6.1-KB2664825-v3-x64.cab")
+      end
+    end
+  end
+
+  describe "#cab_file_source" do
+    it "returns local cab file source path if same is set" do
+      new_resource.source = File.join("#{ENV['TEMP']}", "test6.1-kb2664825-v3-x64.cab")
+      allow(provider).to receive(:cab_file_source).and_return(new_resource.source)
+      provider.load_current_resource
```

Comment:
Please check if it's required to call `load_current_resource` here? Can we directly call `cab_file_source` method since the spec belongs to that method?